code/modules/paperwork/filingcabinet.dm

```diff
@@ -65,20 +65,46 @@
 	else
 		return ..()
 
+/obj/structure/filingcabinet/attack_hand(mob/living/carbon/user, list/modifiers)
+	. = ..()
+	ui_interact(user)
+
+/obj/structure/filingcabinet/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "FilingCabinet")
+		ui.open()
+
+/obj/structure/filingcabinet/ui_data(mob/user)
+	var/list/data = list()
+
+	data["cabinet_name"] = "[name]"
+	data["contents"] = list()
+	data["contents_ref"] = list()
+	for(var/obj/item/content in src)
+		data["contents"] += "[content]"
+		data["contents_ref"] += "[REF(content)]"
 
-/obj/structure/filingcabinet/ui_interact(mob/user)
+	return data
+
+
+/obj/structure/filingcabinet/ui_act(action, params)
 	. = ..()
-	if(contents.len <= 0)
-		to_chat(user, span_notice("[src] is empty."))
+	if(.)
+		return
+
+	if(usr.stat != CONSCIOUS || HAS_TRAIT(usr, TRAIT_HANDS_BLOCKED))
 		return
```

Comment:
shouldn't need this stat check if you are using the default state for objects, it'll do this for you automatically.
You may also want to just move this to ui_status instead of ui_act so that you can also update the status of the UI.

Reply:
You're right. For future reference: https://github.com/tgstation/tgstation/blob/97caf5173877711415afc0404af5315ca3ca7328/code/modules/tgui/states/default.dm does the default `ui_state` checks.